### PR TITLE
Xi precision

### PR DESCRIPTION
**Pull request by MaxThevenet**
xi ~ 1.e-52, which cannot be described by single precision range (smallest number is 1.e-38). Let's use xi*c**2 instead.

### CODE REVIEW

File: Source/FieldSolver/WarpX_QED_Field_Pushers.cpp [hunk 1]
```diff
@@ -152,15 +152,15 @@ WarpX::Hybrid_QED_Push (int lev, PatchType patch_type, Real a_dt)
         );
 
         // Make local copy of xi, to use on device.
-        const Real xi = WarpX::quantum_xi;
+        const Real xi_c2 = WarpX::quantum_xi_c2;
```
Comment (ax3l) [inline Source/FieldSolver/WarpX_QED_Field_Pushers.cpp]: this should be a `constexpr`

File: Source/FieldSolver/WarpX_QED_K.H [hunk 2]
```diff
@@ -144,33 +144,33 @@ const amrex::Real dyi = 1./dy;
...
+        2*xi_c2*( -7*bx*EVxE - 7*VxE[0]*eb + 4*ex*BVxE ) + VxM[0],
+        2*xi_c2*( -7*by*EVxE - 7*VxE[1]*eb + 4*ey*BVxE ) + VxM[1],
+        2*xi_c2*( -7*bz*EVxE - 7*VxE[2]*eb + 4*ez*BVxE ) + VxM[2]
     };
 
     const amrex::Real Omega[3] = {
-        Alpha[0] + 2*xi*c2*( 4*ex*EVxB + 2*VxB[0]*( ee - c2*bb ) + 7*c2*bx*BVxB ),
-        Alpha[1] + 2*xi*c2*( 4*ey*EVxB + 2*VxB[1]*( ee - c2*bb ) + 7*c2*by*BVxB ),
-        Alpha[2] + 2*xi*c2*( 4*ez*EVxB + 2*VxB[2]*( ee - c2*bb ) + 7*c2*bz*BVxB )
+        Alpha[0] + 2*xi_c2*( 4*ex*EVxB + 2*VxB[0]*( ee - c2*bb ) + 7*c2*bx*BVxB ),
```
Comment (ax3l) [inline Source/FieldSolver/WarpX_QED_K.H]: type safety:
```suggestion
        Alpha[0] + 2._rt*xi_c2*( 4._rt*ex*EVxB + 2._rt*VxB[0]*( ee - c2*bb ) + 7._rt*c2*bx*BVxB ),
```

File: Source/FieldSolver/WarpX_QED_Field_Pushers.cpp (hunk 1 above)
Comment (MaxThevenet) [reply Source/FieldSolver/WarpX_QED_Field_Pushers.cpp]: The user can change this parameter at runtime so I don't think it can, right?

**Comment by MaxThevenet** [comment]
Sure, thanks, did it for the int (and some floating-point too).

File: Source/FieldSolver/WarpX_QED_Field_Pushers.cpp (hunk 1 above)
Comment (ax3l) [reply Source/FieldSolver/WarpX_QED_Field_Pushers.cpp]: oh right, mismatched. Sorry.

File: Source/WarpX.cpp [hunk 3]
```diff
@@ -460,7 +460,10 @@ WarpX::ReadParameters ()
         pp.query("n_current_deposition_buffer", n_current_deposition_buffer);
         pp.query("sort_int", sort_int);
 
-        pp.query("quantum_xi", quantum_xi);
+        double quantum_xi;
+        int input_is_specified = pp.query("quantum_xi", quantum_xi);
```
Comment (ax3l) [inline Source/WarpX.cpp]: Maybe that's more specific as a name:
```suggestion
        int quantum_xi_is_specified = pp.query("quantum_xi", quantum_xi);
```

File: Source/WarpX.cpp [hunk 4]
```diff
@@ -460,7 +460,10 @@ WarpX::ReadParameters ()
...
+        if (input_is_specified)
```
Comment (ax3l) [inline Source/WarpX.cpp]:
```suggestion
        if (quantum_xi_is_specified)
```

**Comment by ax3l** [comment]
(Trigger CI)

**Pull request merged**